test/framework/framework.go

```diff
@@ -381,6 +384,17 @@ func (f *Framework) CreateOrUpdatePrometheusOperatorWithOpts(
 	deploy.Spec.Strategy.Type = appsv1.RecreateDeploymentStrategyType
 
 	deploy.Spec.Template.Spec.Containers[0].Args = append(deploy.Spec.Template.Spec.Containers[0].Args, "--log-level=debug")
+	var featureGates string
+	if len(opts.EnabledFeatureGates) > 0 {
+		featureGates = "-feature-gates="
+	}
+	for _, fGate := range opts.EnabledFeatureGates {
+		featureGates += fmt.Sprintf("%s=true,", fGate)
+	}
```

Comment:
(suggestion) to avoid removing the trailing comma when setting the arg list. 
```suggestion
	for i, fGate := range opts.EnabledFeatureGates {
		featureGates += fmt.Sprintf("%s=true", fGate)
		if i < len(range opts.EnabledFeatureGates) - 1 {
			featureGates += ","
		}
	}
```

Reply:
I think that using `featureGates[:len(featureGates)-1]` is more readable 😅

Reply:
no need to change, it was only a suggestion.